### PR TITLE
Improve error handling during reading of TLS configuration.

### PR DESCRIPTION
**Pull request by briansmith**
Switch from panics to "proper" error handling. In particular, now
error messages like "file not found" include the path to the file for
which the I/O was attempted.

### CODE REVIEW

**Comment by briansmith** [comment]
This builds on top of #73 and #74. Only the last (third) commit here is unique to this PR.

**Comment by briansmith** [comment]
In particular, this requires Rustls 0.10, so #74 is required.

File: src/server/sni.rs [hunk 1]
```diff
@@ -3,25 +3,29 @@ use rustls::{Certificate, ResolvesServerCert, SignatureScheme, sign};
...
+    let default = match default {
+        &Some(ref c) => Some(ServerIdentity::load(c)?),
+        &None if n_identities > 0 => {
+            return Err(Error::NoIdentities)
+        },
+        &None => None,
+    };
     let sni = Sni {
-        default: default.as_ref().map(|c| ServerIdentity::load(c)),
+        default: default,
```
Comment (olix0r) [inline src/server/sni.rs]: tioli: recently ruts changed to support the format
```rust
let sni = Sni {
    default,
    ...
```

**Comment by briansmith** [comment]
OK, I fixed the "default: default" thing and rebased. It's good to see that rustc doesn't complain about mixing the abbreviated and non-abbreviated initializer syntax any more.

**Pull request merged**